packages/next/src/client/index.tsx

```diff
@@ -756,6 +765,21 @@ function doRender(input: RenderRouteInfo): Promise<any> {
       looseToArray<HTMLLinkElement>(
         document.querySelectorAll('link[data-n-p]')
       ).forEach((el) => {
+        const href = el.getAttribute('href')
+        if (!href) {
+          return
+        }
+        // When client-navigating to a path which requires the same CSS file,
+        // `mini-css-extract-plugin` will skip injecting the stylesheet as the
+        // link tag with expected href is still in the DOM. Therefore we remove
+        // the `data-n-p` attribute and let the stylesheet remain. This is possible
+        // because we do not "unload" the dynamic stylesheets during the client
+        // navigation already. Therefore the result will be the same with removing
+        // the current stylesheet and then dynamically loading the same stylesheet.
+        if (dynamicCSSFilePaths.has(href)) {
+          el.removeAttribute('data-n-p')
```

Comment:
Could we omit the attribute during SSR directly?

Reply:
Sounds perfect!